powerform/pf-deployment-context/classes/pf-deployment-context.ps1

```diff
@@ -1,17 +1,7 @@
 Class PfDeploymentContext{
-
+    [string]$Name
+    [System.Collections.ArrayList]$BuildContexts=@()
     [System.Collections.ArrayList]$Definitions=@()
-    [PfCloudConfiguration]$AzConfig
-    [PfGlobalConfiguration]$GlobalConfig
-    
-    PfDeploymentContext(){
-        $this.AzConfig=(New-Object -TypeName "PfAzConfiguration")
-        $this.GlobalConfig=(New-Object -TypeName "PfGlobalConfiguration")
-    }
-    [PfStateCollection]Deploy(){
-        return Deploy-PfDeploymentContext
-    }
-    [void]AddDefinition([PfDefinition]$Definition){
-        $this.Definitions.Add($Definition)
-    }
+    [PfBuildContext]$CurrentBuildContext
```

Comment:
Deployment Context now holds an array of BuildContexts that you can add to, update, or switch between within the deployment context.  It also keeps track of the "CurrentBuildContext" that you are working with.